helm/azuredisk-csi-driver-app/templates/azuredisk-controller/deployment.yaml

```diff
@@ -26,7 +27,9 @@ spec:
       nodeSelector:
         kubernetes.io/os: linux
         {{- if .Values.controller.runOnMaster}}
-        kubernetes.io/role: master
+        {{-  range $key, $value := $.Values.controller.masterNodeSelector }}
```

Comment:
this is not working as expected , on CAPZ cluster where we do set these values

https://github.com/giantswarm/default-apps-azure/blob/add_values_json_schema/helm/default-apps-azure/values.yaml#L7

I end up with ( from helm get all azuredisk-csi-driver )

```
  masterNodeSelector:
    kubernetes.io/role: master
    node-role.kubernetes.io/control-plane: ""
```
and

```
      nodeSelector:
        kubernetes.io/os: linux
        kubernetes.io/role: master
        node-role.kubernetes.io/control-plane: 
```

in the actual deployment yaml

which, i guess due to the lack of `""` as a value ends up as 

```
     nodeSelector:
        kubernetes.io/os: linux
        kubernetes.io/role: master
```

in the actual deployment in cluster
which is not what i expected , i only expected the `node-role` one

Reply:
https://github.com/giantswarm/default-apps-azure/pull/7/commits/b4acfb1bdb460df8fcb8ba408cbf397685bf5023

_fixes_ the issue but is quite ugly

Reply:
actually this seems to be the way to this by `nulling` the value in the chart values.yaml 

https://helm.sh/docs/chart_template_guide/values_files/#deleting-a-default-key

unless we restructure the template to something like

```
nodeSelector:
{{- if vintage }}
   kubernetes.io/role: master
{{- else if capz }}
  node-role.kubernetes.io/control-plane: ""
{{ - else }}
  ERROR unsupported 
{{- end }}
```

which would require having `knowledge` of the installation type in the values and would require modifying the template for every possible value of nodeSelector rather than having it configured as a value.

I will keep the fix as it is ( so it does not need any change on vintage values ) but will keep this conversation open for now if anyone want to comment on it